### DISCUSSION TITLE
How many targets are too many?

### DISCUSSION THREAD

**Discussion by nviets** [ropensci/targets#818] [Help]
Silly question - I love dynamic branching and set up a plan to individually track and process thousands of input files, ~100k. Targets does great up to about 3k, and then it really slows down. What are the limiting factors on scale under the hood? Would batching the inputs help (per the section in the dynamic branching chapter)?

**Comment by wlandau** [comment]
Every target has one or more files on disk (usually just one file in `_targets/objects/`), and `targets` records the hash, modification time, and size of each target's file space. These metrics need to be re-computed each time a target reruns. And to check if a target is up to date, the file needs to be checked for existence, then the modification time and size are checked, and then the hash is recomputed if appropriate. This large set of small isolated interactions with the file system is usually the major bottleneck, especially on slow file systems (worst of all Windows mounted network drives). Batching partitions all the existing target output into a smaller number of targets and a smaller number of files in `_targets/objects`, which decreases overhead. There is no set rule for the appropriate number of targets, number of batches, or batch size, but you can experiment and see what works for your setup and resources.

**Reply by nviets** [reply]
Thanks for the clear advice on file systems. Yes, i'm stuck in the worst case scenario with a Windows mounted network drive, and I hadn't diagnosed the issue until you mentioned it. Not that there's anything I can do about it :(
